### PR TITLE
Add support for cluster local tls

### PR DESCRIPTION
**Pull request by dprotaso**
Part of: https://github.com/knative/serving/issues/14375

### Release Note
```
Add support for cluster-local-tls routes
```

### CODE REVIEW

File: pkg/reconciler/contour/resources/httpproxy.go [hunk 1]
```diff
@@ -97,19 +97,31 @@ func defaultRetryPolicy() *v1.RetryPolicy {
 	}
 }
 
+func tlsEntries(ing *v1alpha1.Ingress) map[string]v1alpha1.IngressTLS {
+	external := ing.GetIngressTLSForVisibility(v1alpha1.IngressVisibilityExternalIP)
+	internal := ing.GetIngressTLSForVisibility(v1alpha1.IngressVisibilityClusterLocal)
+
+	entries := make(map[string]v1alpha1.IngressTLS, len(external)+len(internal))
+	for _, tls := range external {
```
Comment (skonto) [inline pkg/reconciler/contour/resources/httpproxy.go:105]: Maybe for more readability you could avoid repeating the loop using a local func and then do:

```
addHostEntries(internal)
addHostEntries(external)
```

**Comment by skonto** [comment]
Some unchanged files have header issues.

**Comment by dprotaso** [comment]
> Some unchanged files have header issues.

I'll do that in a separate PR

**Comment by skonto** [comment]
LGTM waiting for @ReToCode and @izabelacg

**Pull request merged**